src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -63,6 +63,51 @@ public void Kill()
             }
         }
 
+        /// <summary>Stops the associated process immediately.</summary>
+        private void Stop()
+        {
+            EnsureState(State.HaveNonExitedId);
+            if (Interop.Sys.Kill(_processId, Interop.Sys.Signals.SIGSTOP) != 0)
+            {
+                throw new Win32Exception(); // to match behavior of Kill()
+            }
+        }
+
+        /// <summary>
+        /// Makes a best-effort attempt to kill all descendant (child, grandchild, etc.) processes.
+        /// </summary>
+        private void KillTree()
+        {
+            try
+            {
+                /*
+                   Stop but don't kill the process. Keeps additional children from being started but leaves the process around 
+                   so that its children can be enumerated.
+                */
+                Stop();
+            }
+            catch
```

Comment:
please use explicit exception types here

Reply:
I'm glad to add these, if that's what you'd prefer.

Some of the exceptions that will need to be caught may only be thrown in race or difficult to simulate conditions—for example, when a process dies partway through one of the calls this PR makes (maybe after a call to `EnsureState()` checks on it but before an interop call is made on it). It’s somewhere between tricky and impossible to simulate the sequence of events necessary to verify scenarios.

If I code in explicit exception types but don’t have tests to trigger the various scenarios I’m catching exceptions for, I’m concerned that someone might modify code that throws one of those exceptions without realizing they need to modify this PR’s catch code. That other code author likely will walk away thinking things are good (because all tests pass), not realizing that they opened a hole for a one-in-a-million race issue that will show up only in production.  

Here, the exception catch-all bypasses this possibility, albeit with the normal downsides that come with a catch-all. Since the goal for this code is “best attempt, ignore failures,” the catch-all approach looked attractive. Ultimately, though, it boils down to a choice between which approach you feel is better. Just let me know which one you’d prefer (leave things the way they are or add the explicit exception types) and I’ll be glad to use it.

Note: Using a catch-all is done elsewhere in class [`Process`](https://github.com/dotnet/corefx/blob/44e5bdafb8d989a220c9cf1b94f31a64a6e4f052/src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs#L112)

Reply:
We (very?) rarely use catch-all. Your argument is good, but conversely a new exception type might tell us a bug has been introduced. Given that this is a best effort, a middle ground might be to catch concrete exception types, then a catch all - but in the catch all is a Debug.Fail.

Reply:
Sounds good. Working on it.

Reply:
Done